Safe set markers on exceptions (issue #1007)

ayalash opened the pull request.

vmalloc commented on tests/test_exception_marks.py:
```diff
@@ -1,27 +1,41 @@
+
+
+@pytest.fixture(params=[CustomException1, CustomException2])
+def exception_class(request):
```
I'd add some builtin types as well just to make sure

ayalash commented:
@vmalloc I fixed your CR comments. In addition, I added test which use suite-writer to ensure there are no additional failures (`test_add_error_that_forbids_setattr`).

Please review
